src/routes/CommunitySettings/TopicsSettingsTab/TopicsSettingsTab.js

```diff
@@ -0,0 +1,239 @@
+import cx from 'classnames'
+import { find } from 'lodash/fp'
+import { boolean, arrayOf, func, number, shape, string, object } from 'prop-types'
+import React, { Component } from 'react'
+import CreateTopic from 'components/CreateTopic'
+// import { CommunityCell } from 'components/CommunitiesList/CommunitiesList'
+import Dropdown from 'components/Dropdown'
+import Icon from 'components/Icon'
+import SingleTopicSelector from 'components/TopicSelector/SingleTopicSelector'
+import ScrollListener from 'components/ScrollListener'
+import TextInput from 'components/TextInput'
+import { TOPIC_VISIBILITY } from 'store/models/Topic'
+import { inflectedTotal } from 'util/index'
+import styles from './TopicsSettingsTab.scss'
+
+const sortOptions = [
+  { id: 'name', label: 'Name' },
+  { id: 'followersTotal', label: 'Popular' }
+]
+
+const visibilityOptions = Object.keys(TOPIC_VISIBILITY).reduce((result, option) => result.concat({ value: option, label: TOPIC_VISIBILITY[option] }), [])
+
+const TOPIC_LIST_ID = 'topic-list'
+
+export default class TopicsSettingsTab extends Component {
```

Comment:
Curious on this new component if you considered using a functional component and hooks? It's really nice once into it. Perhaps you've already used it elsewhere and are not here in Hylo as it's not in the patterns in this codebase much (at all?) yet?  If I'm creating new components I pretty much always go that way. In looking forward do you have any reasons you don't want to do that?

Reply:
yeah, i do like using functional components moving forward, depending on the complexity of the component, sometimes they don't end up being better. in this case i just copied another tab component. generally yes i agree default to functional!